docs/decisions/0002-java-folder-structure.md

```diff
@@ -0,0 +1,106 @@
+| SemanticKernel.Abstractions    | SK API definitions |
+| SemanticKernel.MetaPackage     | SK common package collection |
+| SemanticKernel.UnitTests       | Unit tests |
+| SemanticKernel                 | SK implementation |
+| Skills                         | Parent folder for various Skills implementations e.g., MS Graph, GRPC, OpenAI, ... |
+
+Some observations:
+
+* The `src` folder is at the very start of the folder structure, which reduces flexibility
+* The use of the `Skills` term is due to change
+
+```bash
+java
+     api-test
+     samples
+     semantickernel-api
+     semantickernel-bom
+     semantickernel-connectors-parent
+     semantickernel-core-skills
+     semantickernel-core
+     semantickernel-extensions-parent
+```
+
+| Folder                              | Description |
+|-------------------------------------|-------------|
+| `api-test`                          | Integration tests and API usage example |
+| `samples`                           | SK samples |
+| `semantickernel-api`                | SK API definitions |
+| `semantickernel-bom`                | SK Bill Of Materials |
+| `semantickernel-connectors-parent`  | Parent folder for various Connector implementations |
+| `semantickernel-core-skills`        | SK core skills (in .Net these are part of the core implementation) |
+| `semantickernel-core`               | SK implementation |
+| `semantickernel-extensions-parent`  | Parent folder for SK extensions e.g., planner implementation |
+
+Some observations:
+
+* Using lowercase folder name with the `-` delimiter is idiomatic Java
+* The `src` folders are located as close as possible to the source files e.g., `semantickernel-api/src/main/java`, this is idiomatic Java
+* Unit tests are contained together with the implementation
+* The samples are located within the `java` folder and each sample runs standalone
+
+## Decision Outcome
+
+Follow these guidelines:
+
+* The folder names will match those used (or planned for .Net) but in the idiomatic Java folder naming convention
+* Use `bom` instead of `MetaPackage` as the latter is .Net centric
+* Use `api` instead of `Abstractions` as the latter is .Net centric
+* Merge `semantickernel-core-skills` and `semantickernel-core` back into a single module name `semantickernel`
+* Use the term `plugins` instead of `skills` and avoid introducing technical debt
+
+| Folder                           | Description |
+|----------------------------------|-------------|
+| `connectors`                     | Containing: `connectors-ai-openai`, `connectors-ai-huggingface`, `connectors-memory-qadrant`, ...  |
+| `extensions`                     | Containing: `planning-action-planner`, `planning-sequential-planner` |
+| `integration-tests`              | Integration tests |
+| `semantickernel-api`             | SK API definitions |
+| `semantickernel-bom`             | SK common package collection |
+| `semantickernel`                 | SK implementation |
```

Comment:
It is idiomatic to keep the module folder as similar as possible (if not the same) of the `artifactId` of said Maven module, with an exception to parent modules (e.g. `connectors` instead of `semantickernel-connectors-parent`).

The module `semantickernel-core`'s folder should probably remain named the same.